### PR TITLE
refactored validate for DA

### PR DESCRIPTION
**Pull request by merit-maita**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-7790

## Description
- Refactored all DA validations.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+FIELDS_TO_INCLUDE = ["fromDate", "toDate", "fromDateLicense"]
+
+
+class IsDashboardContainNecessaryFieldsValidator(BaseValidator[ContentTypes]):
+    expected_git_statuses = [GitStatuses.ADDED]
+    error_code = "DA101"
+    description = "Validate that the dashboard includes all the necessary fields."
+    rationale = "The Dashboard should contains the required 'fromDate', 'toDate', 'fromDateLicense' fields."
+    dashboard_error_message = (
+        "the '{0}' fields are missing from {1} and need to be added."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py:23]:
```suggestion
        "The '{0}' fields are missing from {1} and need to be added."
```

File: demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    )
+    widget_error_message = (
+        "the '{0}' fields are missing from {1} Widget in {2} and need to be added."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py:26]:
```suggestion
        "The '{0}' fields are missing from {1} Widget in {2} and need to be added."
```

File: demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        results = []
+        error_messages = []
+        for content_item in content_items:
+            dashboard_missing_fields = self.dashboard_missing_necessary_fields(
+                content_item.data
+            )
+            widget_missing_fields = self.widget_missing_necessary_fields(
+                content_item.data
+            )
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py:40]: We shouldn't work with the content item data.
We're using graph objects here so the idea is to use the object fields - if necessary then you can add that to the graph object.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Looks good so far, see my comment regarding the content item data.

**Review by YuvHayun — APPROVED** [review]
Looks good!
See my comments.

File: demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    error_code = "DA101"
+    description = "Validate that the dashboard includes all the necessary fields."
+    rationale = "The Dashboard should contains the required 'fromDate', 'toDate', 'fromDateLicense' fields."
+    dashboard_error_message = (
+        "The '{0}' fields are missing from {1} and need to be added."
+    )
+    widget_error_message = (
+        "The '{0}' fields are missing from {1} Widget in {2} and need to be added."
+    )
+    related_field = ""
```
[the remark this answers is not in the document]
Comment (merit-maita) [reply demisto_sdk/commands/validate/validators/DA_validators/DA101_dashboard_contains_necessary_fields.py:28]: in the previous validation, it does not have a related field

File: demisto_sdk/commands/validate/validators/DA_validators/DA100_dashboard_contains_forbidden_fields.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+class IsDashboardContainForbiddenFieldsValidator(BaseValidator[ContentTypes]):
+    expected_git_statuses = [GitStatuses.ADDED]
+    error_code = "DA100"
+    description = "Validate that the dashboard excludes all the unnecessary fields."
+    rationale = "The Dashboard should contains only the required fields."
+    dashboard_error_message = "The '{0}' fields need to be removed from {1}."
+    widgets_error_message = (
+        "The '{0}' fields need to be removed from {1} Widget in {2}."
+    )
+    related_field = ""
```
[the remark this answers is not in the document]
Comment (merit-maita) [reply demisto_sdk/commands/validate/validators/DA_validators/DA100_dashboard_contains_forbidden_fields.py:35]: in the previous validation, it does not have a related field

**Pull request merged**